src/pages/settings/Profile/Contacts/ContactMethodsPage.js

```diff
@@ -0,0 +1,138 @@
+import Str from 'expensify-common/lib/str';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import HeaderWithCloseButton from '../../../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../../../components/ScreenWrapper';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../../../components/withCurrentUserPersonalDetails';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import CONST from '../../../../CONST';
+import compose from '../../../../libs/compose';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import ROUTES from '../../../../ROUTES';
+import LoginField from './LoginField';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Value of partner name */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** Date of when login was validated */
+        validatedDate: PropTypes.string,
+    }),
+
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+const defaultProps = {
+    loginList: {},
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+class ContactMethodsPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            logins: this.getLogins(),
+        };
+
+        this.getLogins = this.getLogins.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        let stateToUpdate = {};
+
+        // Recalculate logins if loginList has changed
+        if (_.keys(this.props.loginList).length !== _.keys(prevProps.loginList).length) {
+            stateToUpdate = {...stateToUpdate, logins: this.getLogins()};
```

Comment:
```suggestion
            stateToUpdate = {logins: this.getLogins()};
```

or just 

```suggestion
            stateToUpdate.logins = this.getLogins();
```

no need to copy and use spread, the object was initialized right above and it is empty

Reply:
Fair and fair - will comment about the general refactoring idea below :D